tests/test_runner.py

```diff
@@ -598,3 +600,105 @@ def test_print_logs_flag_with_iterations():
     assert test_log_pos < test_error_pos
 
     assert ps.stderr == '', Tests.assertion_info('no errors', ps.stderr)
+
+## automatic database reconnection
+# TODO: This integration test should be moved to a dedicated integration test suite once that structure is implemented (https://github.com/green-coding-solutions/green-metrics-tool/issues/1302) # pylint: disable=fixme
+def test_database_reconnection_during_run_integration():
+    """Integration test: Verify GMT runner handles database reconnection during execution
+    
+    This test simulates a database outage by restarting the postgres container mid-run.
+    With database retry logic implemented, the test should succeed by automatically
+    reconnecting when the database becomes available again.
+    
+    Timing analysis based on a debug run with logs:
+    T+0.0s  - GMT runner starts, database restart thread starts waiting
+    T+5.0s  - RUNTIME phase begins (with 15 seconds sleep command)
+    T+12.3s - Database restart occurs (during runtime phase)
+    T+13.1s - Database restart completes
+    T+16.1s - Database should be available again
+    T+20.0s - REMOVE phase starts (database operations resume with retry logic)
+    T+22.4s - Test completes successfully with database reconnection
+    
+    This timing ensures database restart happens during active measurement phase
+    when database operations are likely occurring for metric storage.
+    """
+
+    def log_with_timestamp(message, prefix="TEST", start_time=None):
```

Comment:
if this is a helper function and very general I think placing it in test_functions would be more expected

Reply:
Done.